Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -99,6 +99,37 @@ extension BrowserViewController: WKNavigationDelegate {
             decisionHandler(.cancel)
             return
         }
+        
+        let isSafeBrowsingEnabled = { () -> Bool in
+            let isPrivateBrowsing = PrivateBrowsingManager.shared.isPrivateBrowsing
+            let domain = Domain.getOrCreate(forUrl: url, persistent: !isPrivateBrowsing)
+            return domain.isShieldExpected(.SafeBrowsing)
+        }
+        
+        if isSafeBrowsingEnabled() {
+            var isURLSafe = true //By default, we mark the URL as safe, unless the lookup returns otherwise..
+            let semaphore = DispatchSemaphore(value: 0)
+            DispatchQueue.global(qos: .background).async {
+                SafeBrowsingClient.shared.find(url.hashPrefixes()) { isSafe, error in
+                    defer { semaphore.signal() }
+                    
+                    if let error = error {
+                        log.error(error)
+                        return
+                    }
+                    
+                    isURLSafe = isSafe
+                }
+            }
+            
+            _ = semaphore.wait(timeout: .now() + .seconds(30))
+            
+            if !isURLSafe {
+                safeBrowsing?.showMalwareWarningPage(forUrl: url, inWebView: webView)
```

Comment:
Where is `showMalwareWarningPage()` implemented? Does it make a difference between malware, phishing and unwanted software?

Reply:
It's a page I just decided to show which already existed long before my time, when a URL was blocked. It doesn't make any distinction for any results. I don't have requirements or design for this feature.

Reply:
Ah I see, it was for the [old `SafeBrowsing.dat` file](https://github.com/brave/ad-block/blob/c02a65217e5e989294fef06ce4a2e099c9c10c4b/lists/malware.h) (nothing to do with Google Safe Browsing).

I guess we'll to design our own warning pages for this, based on the [official guidelines](https://developers.google.com/safe-browsing/v4/usage-limits#UserWarnings).